L1Trigger/L1TTwinMux/src/AlignTrackSegments.cc

```diff
@@ -28,42 +28,53 @@ void AlignTrackSegments::run( const edm::EventSetup& c) {
 
   std::vector<L1MuDTChambPhDigi> l1ttma_out;
 
-  L1MuDTChambPhDigi const* ts1=0;
-  L1MuDTChambPhDigi const* ts2=0;
+ // L1MuDTChambPhDigi const* ts1=0;
+//  L1MuDTChambPhDigi const* ts1=0;
+ // L1MuDTChambPhDigi const* ts2=0;
+//  std::auto_ptr<const L1MuDTChambPhDigi> ts1(new L1MuDTChambPhDigi);
+ // std::unique_ptr<L1MuDTChambPhDigi> ts2(new L1MuDTChambPhDigi);
 
+  auto ts1=std::make_shared<L1MuDTChambPhDigi>(0,0,0,0,0,0,7,0,0,0);
+  auto ts2= std::make_shared<L1MuDTChambPhDigi>(0,0,0,0,0,0,7,0,0,0);
   int bx=0, wheel=0, sector=0, station=1;
 
 
-for(bx=-3; bx<=3; bx++){
-for (wheel=-3;wheel<=3; wheel++ ){
-	for (sector=0;sector<12; sector++ ){
-     for (station=1; station<=4; station++){
+  for(bx=-3; bx<=3; bx++){
+   for (wheel=-3;wheel<=3; wheel++ ){
+    for (sector=0;sector<12; sector++ ){
+      for (station=1; station<=4; station++){
         ///chPhiSegm1 reads the first ts from bx
-        ts1 = m_phiDigis.chPhiSegm1(wheel,station,sector,bx);
+        ts1.reset(new L1MuDTChambPhDigi(0,0,0,0,0,0,7,0,0,0));
```

Comment:
This will probably give a bad performance since new L1MuDTChampPhDigis are constructed all the time.
Better to keep the raw pointers in this case.

Reply:
This is also my opinion. Smart pointer is not suited. Also it is very error prone since in some files we need to use it 2-3 times